client/src/components/Input/TextChat.jsx

```diff
@@ -28,7 +29,16 @@ export default function TextChat({ isSearchView = false }) {
   const { ask, stopGenerating } = useMessageHandler();
   const isNotAppendable = latestMessage?.unfinished & !isSubmitting || latestMessage?.error;
   const { conversationId, jailbreak } = conversation || {};
+  const { isSpeechSupported, isListening, text: speechText } = useSpeechRecognition(ask);
 
+  useEffect(() => {
```

Comment:
Probably nothing you changed, but need to add: I can click the microphone icon while it's listening and doing so will refresh the page